tensorflow/python/framework/python_op_gen.cc

```diff
@@ -713,14 +713,24 @@ string GetPythonOp(const OpDef& op_def, const string& function_name) {
 }
 
 string GetPythonOps(const OpList& ops, const std::vector<string>& hidden_ops,
-                    bool require_shapes) {
+                    const string& source_file_name, bool require_shapes) {
   string result;
   // Header
   // TODO(josh11b): Mention the library for which wrappers are being generated.
   strings::StrAppend(&result, R"("""Python wrappers around TensorFlow ops.
 
 This file is MACHINE GENERATED! Do not edit.
-"""
+)");
+ 
+  // Mention the original source file so someone tracing back through generated
+  // Python code will know where to look next.
+  if (source_file_name.size() > 0) {
```

Comment:
The idiom we use through this code base is to say
if (!source_file_name.empty())

Reply:
Sorry, didn't notice this review comment the last time I checked up on the PR. Thanks for the reminder! I've fixed my local copy. Will push the changes once I've merged updates from the trunk and run tests.